ui/raidboss/raidemulator.js

```diff
@@ -0,0 +1,832 @@
+'use strict';
+
+// TODO: allow for different playback rates.  The main trickiness
+// here is that (1) timeline playback needs to be adjusted and
+// (2) delaySeconds/suppressSeconds also need to be fixed.
+
+// TODO: make it possible to scrub around forwards and backwards
+// in time in the player.  Forwards is easiest, because the
+// player could just play all the logs (silently).  Backwards
+// is much harder because it would need to start from the
+// beginning again (or some data checkpoint).
+
+// TODO: allow the emulator to suppress events from the game
+// such as zone changing or player data changing in raidboss.
+// Currently, switching zones will stop the emulator.
+
+// TODO: it would be nice to let the user select their name
+// and role to pretend to be somebody else in the log.  The user
+// could input this manually, or maybe select from the actors
+// detected in the fight.  Although this would require detecting what
+// jobs they were, probably by matching on particular skills that
+// they likely would use, e.g. / 15:........:name:1F:/ for heavy swing.
+
+let gLogCollector;
+let gEmulatorView;
+
+// Responsible for handling importing the fights.
+// When a fight is imported, it will call the addFightCallback.
+class LogCollector {
+  constructor(addFightCallback) {
+    this.currentZone = null;
+    // { key:, zoneName:, startDate:, endDate:, durationMs:, logs: }
+    this.fights = [];
+    this.currentFight = null;
+    this.addFightCallback = addFightCallback;
+    this.wait;
+  }
+
+  StoreFights(fights) {
+    localStorage.clear();
+    localStorage.setItem('fights', JSON.stringify(fights));
+  }
+
+  RestoreOldFights() {
+    if (localStorage.fights)
+      this.fights = JSON.parse(localStorage.getItem('fights'));

```

Comment:
If I try to even click on a ucob fight from my log, I get this error: `12/30/2018 10:30:25 PM: Info: emulator: BrowserConsole: Uncaught QuotaExceededError: Failed to execute 'setItem' on 'Storage': Setting the value of 'fights' exceeded the quota. (Source: file:///C:/Users/enne/cactbot/ui/raidboss/raidemulator.js, Line: 41)`

I agree that it's a huge pain to have to keep importing, but I'm also skeptical about this storage method given these errors.

Reply:
First thing I thought of were cookies but since they are disabled I went for the localStorage which is pretty limited. Maybe it would help to store just the selected fight?